client/src/partials/inventory/list/modals/search.modal.js

```diff
@@ -0,0 +1,59 @@
+angular.module('bhima.controllers')
+.controller('InventoryServiceModalController', InventoryServiceModalController);
+
+InventoryServiceModalController.$inject = [
+  '$uibModalInstance', 'params', 'InventoryService'
+];
+
+/**
+ * @class InventoryServiceModalController
+ *
+ * @description
+ * This controller is responsible for setting up the filters for the patient
+ * search functionality on the patient registry page.  Filters that are already
+ * applied to the grid can be passed in via the params inject.
+ */
+function InventoryServiceModalController(ModalInstance, params, InventoryGroups) {
+  var vm = this;
+
+  // bind filters if they have already been applied.  Otherwise, default to an
+  // empty object.
+  vm.params = params || {};
+
+  // bind methods
+  vm.submit = submit;
+  vm.cancel = cancel;
+  vm.clear = clear;
+
+  InventoryGroups.Groups.read()
```

Comment:
The service used here is actually the `InventoryService`.  Renaming it `InventoryGroups` is misleading.  Please change the name so that this code reads:

```js
Inventory.Groups.read()
```